### ISSUE TITLE
[explicit-member-accessibility] option to support leaving accessibility modifiers off of class member overrides

### ISSUE THREAD

**Issue by mattlohkamp** [typescript-eslint/typescript-eslint#4246]
### Description

I'd like to request the following option:

If a member is marked 'protected' in a base Class, when that Class is extended, an override of that member should not require an explicit accessibility modifier - that member's accessibility is already implicit, inherited from the base Class it was defined in.

### Fail

```ts
class Class1 { protected prop1 = 'foo'; }
class Class2 extends Class1 { protected override prop1 = 'bar'; }
```
there's no need to redeclare member `foo` as protected

### Pass

```ts
class Class1 { protected prop1 = 'foo'; }
class Class2 extends Class1 { override prop1 = 'bar'; }
class Class3 extends Class1 { public override prop1 = 'baz'; }
```
if accessibility is modified on the override, that's fine

**Comment by bradzacher** [comment]
Unfortunately this is not possible because it would require type information - which the rule does not use.
